### ISSUE TITLE
Cursor Stays at Start of the line in text fields and Ignores new line characters characters

### ISSUE THREAD

**Issue by IamTalhaAhmed** [flutter/flutter#145656]
### Steps to reproduce

1. Using fvm made my flutter version 3.7.10
2. Adjusted dependencies and sdk version according to that
3. Run the project on IPhone 15 pro max. (it occured on IPhone 14 Promax as well).i guess its not related to device.
This issue is not just here actually its the details of the project which was working fine earlier with flutter 3.10.6 but then just to reproduce I downgraded the version to 3.7.10 and this issue started happening which is actually happening in one of my client's flutter project

### Expected results

Cursor should move forward with each character entered.
It should identify the \n characters for next line

### Actual results

Wherever I am using some type of text in my project it is having the issues. 
1. For text input fields the cursor stays at start regardless of how many characters I have entered.
2. Enter (\n) is getting treated as a single space.

### Code sample

<details open><summary>Code sample</summary>

```dart
TextField(
        controller:usernameController,
        obscureText: false,
        decoration: InputDecoration(
            enabledBorder: OutlineInputBorder(
              borderSide: BorderSide(color: Colors.white),
              borderRadius: BorderRadius.circular(12)
            ),
            focusedBorder: OutlineInputBorder(
              borderSide: BorderSide(color: Colors.grey.shade400),
            ),
            fillColor: Colors.grey.shade200,
            filled: true,
            hintText: 'Email',
            hintStyle: TextStyle(
              color: Colors.grey[500]
            )
        ),
      ),
```

</details>

### Screenshots or Video

https://github.com/flutter/flutter/assets/112610571/b790fd2e-013f-4fca-9e07-9ebe5e6ffc8b

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
flutter doctor -v
[✓] Flutter (Channel stable, 3.10.6, on macOS 14.2.1 23C71 darwin-x64, locale en-PK)
    • Flutter version 3.10.6 on channel stable at /Users/talha/tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (9 months ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/talha/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio 2.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.86.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.84.0

[✓] VS Code (version 1.80.1)
    • VS Code at /Users/talha/Downloads/dmgs/Visual Studio Code.app/Contents
    • Flutter extension version 3.84.0

[✓] Connected device (3 available)
    • iPhone 15 Pro Max (mobile) • 7B303682-5A8E-41F5-B062-83807B5BC61C • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-4 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • macOS 14.2.1 23C71 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 123.0.6312.59

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```
fvm --version
2.4.1
```

```
fvm doctor

FVM Version: 2.4.1
___________________________________________________

FVM config found:
___________________________________________________

Project: car_management
Directory: /Users/talha/work/Flutter-projects/CarManagement/car_management
Version: 3.7.10
Project Flavor: None selected
___________________________________________________

Version is currently cached locally.

Cache Path: /Users/talha/fvm/versions/3.7.10
Channel: false
SDK Version: 3.7.10

IDE Links
VSCode: .fvm/flutter_sdk
Android Studio: /Users/talha/work/Flutter-projects/CarManagement/car_management/.fvm/flutter_sdk


Configured env paths:
___________________________________________________

Flutter:
/Users/talha/tools/flutter/bin/flutter

Dart:
/Users/talha/tools/flutter/bin/dart

FVM_HOME:
not set
```

```
fvm list
Cache Directory:  /Users/talha/fvm/versions

3.10.6
3.7.10 (active)
```

```
fvm flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel stable, 3.7.10, on macOS 14.2.1 23C71 darwin-x64, locale en-PK)
    ! Warning: `flutter` on your path resolves to /Users/talha/tools/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/talha/fvm/versions/3.7.10. Consider adding /Users/talha/fvm/versions/3.7.10/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/talha/tools/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/talha/fvm/versions/3.7.10. Consider adding /Users/talha/fvm/versions/3.7.10/bin to the front of your path.
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.86.2)
[✓] VS Code (version 1.80.1)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

! Doctor found issues in 1 category.
```

```
fvm flutter --version
Flutter 3.7.10 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 4b12645012 (12 months ago) • 2023-04-03 17:46:48 -0700
Engine • revision ec975089ac
Tools • Dart 2.19.6 • DevTools 2.20.1
```

</details>

**Comment by huycozy** [comment]
Hi @IamTalhaAhmed 
I checked this issue using minimal sample code below, on the Flutter latest stable channel 3.19.4 but don't see the issue. Your Flutter version is quite old, please try switching to 3.19.4 and confirm.

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        body: Center(
          child: TextField(
            obscureText: false,
            decoration: InputDecoration(
              enabledBorder: OutlineInputBorder(
                borderSide: const BorderSide(color: Colors.white),
                borderRadius: BorderRadius.circular(12),
              ),
              focusedBorder: OutlineInputBorder(
                borderSide: BorderSide(color: Colors.grey.shade400),
              ),
              fillColor: Colors.grey.shade200,
              filled: true,
              hintText: 'Email',
              hintStyle: TextStyle(color: Colors.grey[500]),
            ),
          ),
        ),
      ),
    ),
  );
}

```
</details>

**Comment by IamTalhaAhmed** [comment]
Hi @huycozy. the project has been built using that old version so all the dependencies and packages are bound to that as well so upgrading a flutter version will be fine?

**Comment by IamTalhaAhmed** [comment]
it will be a great favor if you switch to this version and can check the cause of this issue. As for this sample project I was using flutter 3.10.6 earlier and it was working fine just while trying to reproduce the issue in my real project I tried to use the old flutter version for this project as well and I landed to the same issue.

**Comment by huycozy** [comment]
Unfortunately, I don't store many old Flutter versions as my machine has low storage. `3.10.6` is quite an old version and there have been many commits since then. The issue could be caused by implementation or a framework bug that could be fixed. I would suggest you check with the minimal sample code I shared above first on your Flutter version 3.10.6. 

If the issue occurs on 3.10.6, there must be a fix since then until the current stable 3.19.4. Otherwise, it could be an implementation issue in the project, and you should narrow down the issue by isolating the affected page and finding the cause.

**Comment by IamTalhaAhmed** [comment]
Its Fixed by upgrading the Flutter version.
